src/PKSim.Core/Services/TimeProfileChartsDataCreator.cs

```diff
@@ -48,6 +50,52 @@ protected override bool CheckFields()
          return _analysis.AllFieldsOn(PivotArea.DataArea).Any() && statisticalAnalysis.SelectedStatistics.Any();
       }
 
+      public override IEnumerable<PopulationPKAnalysis> Aggregate(IEnumerable<StatisticalAggregation> selectedStatistics, IReadOnlyList<Compound> compounds, IEnumerable<QuantityPKParameter> pkParameters, Simulation simulation, string captionPrefix)
+      {
+         var names = pkParameters.Select(x => x.Name).Distinct();
+         var matrix = new FloatMatrix();
+         pkParameters.Each(pkParameter => matrix.AddValuesAndSort(pkParameter.ValuesAsArray));
+
+         var results = new List<PopulationPKAnalysis>();
+         selectedStatistics.Each(statisticalAnalysis => {
+            var aggregated = _statisticalDataCalculator.StatisticalDataFor(matrix, statisticalAnalysis).ToList();
+            for (var aggregationIndex = 0; aggregationIndex < aggregated.Count; aggregationIndex++)
+            {
+               var pk = pkParameters.ElementAt(aggregationIndex);
+               var curveData = buildCurveData(pk, aggregated, aggregationIndex, statisticalAnalysis, captionPrefix);
+               results.Add(buildPopulationPKAnalysis(curveData, compounds.First(x => curveData.Caption.Contains(x.Name)), aggregated[aggregationIndex], names, simulation));
+            }
+         });
+         return results;
+      }
+
+      private CurveData<TimeProfileXValue, TimeProfileYValue> buildCurveData(QuantityPKParameter pk, List<float[]> values, int index, StatisticalAggregation statisticalAggregation, string captionPrefix)
+      {
+         var suffix = _representationInfoRepository.DisplayNameFor(statisticalAggregation);
+         if (values.Count > 1)
+            suffix = index == 0 ? PKSimConstants.PKAnalysis.LowerSuffix(suffix) : PKSimConstants.PKAnalysis.UpperSuffix(suffix);
+         var caption = captionFor(new[] { captionPrefix, suffix });
+
+         var curveData = new CurveData<TimeProfileXValue, TimeProfileYValue>()
+         {
+            Id = pk.Id,
+            Caption = caption,
+            YDimension = pk.Dimension,
+            QuantityPath = pk.QuantityPath,
+         };
+         return curveData;
+      }
+
+      private PopulationPKAnalysis buildPopulationPKAnalysis(CurveData<TimeProfileXValue, TimeProfileYValue> curveData, Compound compound, float[] values, IEnumerable<string> names, Simulation simulation)
```

Comment:
make name an IReadOnly here so that you don't have to use Count() and ElementAt() etc..